java/java.hints.declarative/src/org/netbeans/modules/java/hints/declarative/DeclarativeHintRegistry.java

```diff
@@ -229,8 +229,12 @@ private static Collection<? extends FileObject> findFilesRecursive(FileObject fo
 
     public static Map<HintMetadata, Collection<? extends HintDescription>> parseHintFile(@NonNull FileObject file) {
         String spec = Utilities.readFile(file);
-
-        return spec != null ? parseHints(file, spec) : Collections.emptyMap();
+        try {
+            return spec != null ? parseHints(file, spec) : Collections.emptyMap();
+        } catch (Exception ex) {
+            LOG.log(Level.WARNING, "Can not parse hint file: "+file.toURL()+"\n"+spec, ex);
+            return Collections.emptyMap();
+        }
```

Comment:
I think this is the wrong place to catch the exception. For example in `HintsActionProvider#invokeAction` could also just show info to the user, instead of reporting that "no hints" are present and giving the real info by an exclamation mark in the status bar.

`TestPerformer#perform`  most probably should report a broken test as test error.

For `DeclarativeHintRegistry` it might be the right approach - not sure how it is really used.

Reply:
`~/.netbeans/[version]/conf/rules ` are put into the "Custom" category. You can edit the files and it will update on save in real time.

This ensures that a single invalid file does not break everything and disable hints.

Reply:
also worth mentioning that the hint file itself would usually display parser errors, however the parser doesn't like non trivial hint files and it is typically full with error badges (and parsing exceptions). I tried to find ways some time ago to disable semantic editor features for hint files but gave up at some point.

Reply:
@matthiasblaesing i reverted this change